__tests__/app.test.js

```diff
@@ -39,39 +39,143 @@ describe('/api/categories', () => {
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            designer: expect.any(String),
+                            comment_count: expect.any(Number),
+                        })
+                    );
+                    expect({ review }).toEqual(
+                        expect.not.objectContaining({ review_body: expect.anything() })
+                    );
+                });
+            });
+    });
 
-        test('should return an array sorted by created date', () => {
-            return request(app)
-                .get("/api/reviews")
-                .expect(200)
-                .then(({ body }) => {
-                    expect(body.reviews).toBeSortedBy('created_at', { descending: true, coerce: true, })
-                })
-        })
+    test("should return a filtered reviews array if the category parameter is given", () => {
+        return request(app)
+            .get("/api/reviews?category=social deduction")
+            .expect(200)
+            .then(({ body }) => {
+                const { reviews } = body;
+                expect(reviews).toHaveLength(11);
+                reviews.forEach((review) => {
+                    expect(review).toEqual(
+                        expect.objectContaining({
+                            owner: expect.any(String),
+                            title: expect.any(String),
+                            review_id: expect.any(Number),
+                            category: 'social deduction',
+                            review_img_url: expect.any(String),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            designer: expect.any(String),
+                            comment_count: expect.any(Number),
+                        })
+                    );
+                    expect({ review }).toEqual(
+                        expect.not.objectContaining({ review_body: expect.anything() })
+                    );
+                });
+            });
+    });
+
+    test("should return a 400 status code if an invalid sort_by query is entered", () => {
+        return request(app)
+            .get("/api/reviews?sort_by=a")
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toEqual("Bad Request");
+            });
+    });
+
+    test("should return a 400 status code if an invalid order_by query is entered", () => {
+        return request(app)
+            .get("/api/reviews?order=a")
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toEqual("Bad Request");
+            });
+    });
+
+    test("should return a 400 status if a category that doesn't exist is entered", () => {
+        return request(app)
+            .get("/api/reviews?category=nonExistentCategory")
+            .expect(400)
```

Comment:
This should be a 404 as the category doesn't exist.... but could do in the future!